lib/internal/main/watch_mode.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+const {
+  ArrayPrototypeJoin,
+  ArrayPrototypeMap,
+  ArrayPrototypePush,
+  ArrayPrototypePushApply,
+  ArrayPrototypeReduce,
+  ArrayPrototypeSlice,
+} = primordials;
+const {
+  prepareMainThreadExecution,
+  markBootstrapComplete
+} = require('internal/process/pre_execution');
+const { getOptionValue } = require('internal/options');
+const { emitExperimentalWarning } = require('internal/util');
+const { FilesWatcher } = require('internal/watch_mode/files_watcher');
+const { green, blue, red, white } = require('internal/util/colors');
+
+const { spawn } = require('child_process');
+const { inspect } = require('util');
+const { setTimeout, clearTimeout } = require('timers');
+const { resolve } = require('path');
+const { once, on } = require('events');
+
+
+prepareMainThreadExecution(false, false);
+markBootstrapComplete();
+
+// TODO(MoLow): Make kill signal configurable
+const kKillSignal = 'SIGTERM';
+const kShouldFilterModules = getOptionValue('--watch-path').length === 0;
+const kWatchedPaths = getOptionValue('--watch-path').length ?
+  ArrayPrototypeMap(getOptionValue('--watch-path'), (path) => resolve(path)) :
+  [process.cwd()];
+const kCommand = ArrayPrototypeSlice(process.argv, 1);
+const kCommandStr = inspect(ArrayPrototypeJoin(kCommand, ' '));
+const args = ArrayPrototypeReduce(process.execArgv, (acc, flag, i, arr) => {
```

Comment:
Kind of a weird way to filter?

Reply:
this filters pairs of items:
i.e this is an expected result,
`['a', '--flag', 'value', 'b'] ==> ['a', 'b']` 
but trying to avoid
`['--flag', 'a', 'value', 'b'] ==> ['a', 'b']`